src/ddscxx/include/org/eclipse/cyclonedds/core/cdr/basic_cdr_ser.hpp

```diff
@@ -253,8 +253,11 @@ void read_string(basic_cdr_stream& str, T& toread, size_t N)
 
   read(str, string_length);
 
+  //the string length in the CDR stream includes the terminating NULL
+  //therefore the checks on the string length decrease it by 1
+
   if (N &&
-      string_length > N &&
+      string_length - 1 > N &&
```

Comment:
with an empty string, doesn't the `string_length - 1` lead to some large unintended value?

Reply:
It may seem that way, but this won't happen, because the length field read from the CDR stream includes the terminating NULL character (CORBA v3.3 section 9.3.2.7), unlike the value from std::string::length.